extensions/eclipse-che-theia-plugin-remote/src/browser/plugin-resource.ts

```diff
@@ -0,0 +1,66 @@
+/**********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+import { MaybePromise, Resource, ResourceReadOptions, ResourceResolver } from '@theia/core/lib/common';
+
+import { Endpoint } from '@theia/core/lib/browser';
+import { PluginUri } from '../common/plugin-uri';
+import URI from '@theia/core/lib/common/uri';
+import { injectable } from 'inversify';
+
+/**
+ * A resource for loading plugin resources. Uris will be of the form
+ * pluginresource://<plugin id>/path/relative/to/plugin/location
+ * The path is resolve relative to the directory designated by the
+ * plugins package path.
+ */
+export class PluginResource implements Resource {
```

Comment:
Should be a ChePluginResource

Reply:
No, it's not Che specific, even if it lives inside che-theia: same reasoning as above. I'm open to proposing a PR upstream with the plugin resource stuff.

Reply:
if it's becoming an upstream yes it'll be a pluginResource else it'll be a Che plugin resource.
Proposing it upstream will be better if you want to stick to 'pluginResource'

Reply:
But I think it's only useful upstream if this protocol is used somewhere